src/sqlfluff/dialects/dialect_spark3.py

```diff
@@ -1109,3 +1109,108 @@ class JoinClauseSegment(BaseSegment):
     get_eventual_alias = ansi_dialect.get_segment(
         "JoinClauseSegment"
     ).get_eventual_alias
+
+
+@spark3_dialect.segment(replace=True)
+class AliasExpressionSegment(BaseSegment):
+    """A reference to an object with an `AS` clause.
+
+    The optional AS keyword allows both implicit and explicit aliasing.
+    Note also that it's possible to specify just column aliases without aliasing the
+    table as well:
+    .. code-block:: sql
+
+        SELECT * FROM VALUES (1,2) as t (a, b);
+        SELECT * FROM VALUES (1,2) as t;
+        SELECT * FROM VALUES (1,2) as (a, b);
+
+    """
+
+    type = "alias_expression"
+    match_grammar = Sequence(
+        Ref.keyword("AS", optional=True),
+        OneOf(
+            Sequence(
+                # maybe table alias
+                Ref("SingleIdentifierGrammar", optional=True),
+                # maybe column aliases
+                Bracketed(Ref("SingleIdentifierListSegment")),
+            ),
+            # maybe table alias
+            Ref("SingleIdentifierGrammar"),
```

Comment:
Should we also support `SingleQuotedIdentifierSegment`, as seen in the ANSI dialect?

Reply:
hm, I am not grokking what that would be, do you know an example that would cover it?

`SingleIdentifierGrammar` should cover both naked and quoted identifiers, which in spark3 means single backticks

Reply:
In ANSI we have `SingleIdentifierGrammar` which is `NakedIdentifierSegment` (e.g. `barry`) or `QuotedIdentifierSegment` (e.g. `"barry"` in most dialects or with backticks in spark3).

However there is also a `SingleQuotedIdentifierSegment` (e.g. `'barry'`) which is only allowed to be used in some places, whereas `QuotedIdentifierSegment`s are allowed to be used in a lot more places. Therefore `SingleQuotedIdentifierSegment` it is not part of `SingleIdentifierGrammar` and needs to be explicitly added to the few places it's acceptable.

The Alias is one of those special places that allows this as well as normal quoted identifiers. Therefore it is explicitly added there in ANSI:

```py
    type = "alias_expression"
    match_grammar = Sequence(
        Ref.keyword("AS", optional=True),
        OneOf(
            Sequence(
                Ref("SingleIdentifierGrammar"),
                # Column alias in VALUES clause
                Bracketed(Ref("SingleIdentifierListSegment"), optional=True),
            ),
            Ref("SingleQuotedIdentifierSegment"),
        ),
    )
```

So @barrywhart 's question is, should it be here too? If not, (because spark3 only allows backticks quotes even here) it might be good to add a comment to that effect as it's different from ANSI definition.

Reply:
i see, let me verify the behavior and i'll either amend to cover the single quoted case or note the variance from ANSI

Reply:
it's not allowed, i've called this out in the docstring